tools/review_pull_request.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python
+# License AGPLv3 (http://www.gnu.org/licenses/agpl-3.0-standalone.html)
+
+import argparse
+from collections import Counter
+from datetime import datetime, timedelta
+
+from github_login import login
+
+from oca_projects import OCA_USERNAME, OCA_REPOSITORY_NAMES
+
+contributions_cache = None
+
+
+def get_contribution_counts():
+    global contributions_cache
+
+    if contributions_cache is not None:
+        return contributions_cache
+
+    github = login()
+    contributions_cache = Counter()
+
+    for repository_name in OCA_REPOSITORY_NAMES:
+        repository = github.repository(OCA_USERNAME, repository_name)
+        for user in repository.contributors():
+            contributions_cache[user.id] += user.contributions_count
+
+    return contributions_cache
+
+
+def tag_new_pull_requests(repository_name):
+    github = login()
+    repository = github.repository(OCA_USERNAME, repository_name)
+    contributions = get_contribution_counts()
+
+    for pull_request in repository.pull_requests():
+        if contributions[pull_request.user.id] < 3:
+            pull_request.issue().add_labels('new contributor')
+
+
+def tag_all_new_pull_requests():
+    for repository_name in OCA_REPOSITORY_NAMES:
+        tag_new_pull_requests(repository_name)
+
+
+def close_abandoned_pull_requests(repository_name):
+    github = login()
+    repository = github.repository(OCA_USERNAME, repository_name)
+
+    for pull_request in repository.pull_requests():
+        full_pull_request = repository.pull_request(pull_request.number)
+
+        now = datetime.now(pull_request.created_at.tzinfo)
+        pr_is_old = now - pull_request.created_at > timedelta(days=31 * 6)
+
+        if pr_is_old and full_pull_request.comments_count == 0:
+            pull_request.create_comment("Please re-open if necessary")
```

Comment:
We can put here a longer text for not being rude, like:
```
This is an automatic message for telling you that this PR is going to be closed due to a long inactivity period.

Please ask for reopening if needed.
"""

Reply:
Just a nitpicking comment here: isn't better to use the last update date? If some big PR is having a long discussion it will be auto-closed after 6 months even if there is recent conversations/reviews.

Reply:
Then the comments_count could be removed